components/psst/browser/core/psst_rule_service.h

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_COMPONENTS_PSST_BROWSER_CORE_PSST_RULE_SERVICE_H_
+#define BRAVE_COMPONENTS_PSST_BROWSER_CORE_PSST_RULE_SERVICE_H_
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "base/component_export.h"
+#include "base/memory/raw_ptr.h"
+#include "base/memory/singleton.h"
+#include "base/memory/weak_ptr.h"
+#include "brave/components/psst/browser/core/psst_rule.h"
+#include "third_party/abseil-cpp/absl/types/optional.h"
+
+class GURL;
+
+namespace psst {
+
+class COMPONENT_EXPORT(PSST) PsstRuleService {
+ public:
+  PsstRuleService(const PsstRuleService&) = delete;
+  PsstRuleService& operator=(const PsstRuleService&) = delete;
+  ~PsstRuleService();
+  COMPONENT_EXPORT(PSST)
```

Comment:
Nit: I suspect we don't need these `COMPONENT_EXPORT(PSST)` declarations here, as we've already declared the class itself with it. Maybe I'm wrong.

Reply:
well, in this case I think you might actually want to do it the other way around so that only the public api methods are exported, but this can cause some confusion when building shared vs static libs so it's better to do an interface/impl or make `LoadRules` private and add PsstComponentInstallerPolicy as a friend